### DISCUSSION TITLE
SMA update, Y-axis display  problem

### DISCUSSION THREAD

**Discussion by youmeomeyou** [highfestiva/finplot#190] [Q&A] [answered]
Here is my code:
# Data part-------------------------
self.fplt_price, self.fplt_volume = fplt.create_plot(tiker, init_zoom_periods=100, rows=2)

self.verticalLayout_chart.addWidget(self.fplt_price.vb.win)
self.verticalLayout_chart.addWidget(self.fplt_volume.vb.win)

self.chart_candle = fplt.candlestick_ochl(self.chart_data[['open', 'close', 'high', 'low']], ax=self.fplt_price, candle_width=0.6)
self.chart_ma20 = fplt.plot(self.chart_data['close'].rolling(20).mean(), ax=self.fplt_price, legend='SMA 20')
self.chart_ma5 = fplt.plot(self.chart_data['close'].rolling(5).mean(), ax=self.fplt_price, legend='SMA 5')
self.chart_ma3 = fplt.plot(self.chart_data['close'].rolling(3).mean(), ax=self.fplt_price, legend='SMA 3')
self.chart_volume = fplt.volume_ocv(self.chart_data[['open', 'close', 'value']], ax=self.fplt_volume, candle_width=0.6)

# Update Part----------------------
self.chart_ma20.update_data(self.chart_data['close'].rolling(20).mean(), gfx=False)
self.chart_ma5.update_data(self.chart_data['close'].rolling(5).mean(), gfx=False)
self.chart_ma3.update_data(self.chart_data['close'].rolling(3).mean(), gfx=False)

self.chart_candle.update_data(self.chart_data[['open', 'close', 'high', 'low']], gfx=False)
self.chart_volume.update_data(self.chart_data[['open', 'close', 'value']], gfx=False)

fplt._repaint_candles()
#------------------------------------
[https://drive.google.com/file/d/1uv5vQM2WpkU6kXy7xSzokwaKRwtP9RnN/view?usp=sharing](url)

The candlechart is being updated.
But SMA is not.

How can I update it?

and Y-axis Data is float. but I want to be seen as int.
How can I change view?

Thank you for read

**Comment by highfestiva** [comment, marked as answer]
Please post a complete but minimal example and I'll have a look. I don't have time to compose and complete half-made examples.

See [custom axes ticks](https://github.com/highfestiva/finplot#custom-axes-ticks) for more info on that.

**Reply by youmeomeyou** [reply]
First of all, thank you for your great finplot.
This solved my many months of trouble.
Almost everything is satisfactory.

You are excellent, and I think you will be very busy.

So I uploaded only the simple code part as above.

I just want to know the method.

_repaint_candles() updated the stick chart.
However, SMA could not be updated.
Which method causes the SMA to be updated?

And

Y-axis type float.
Y-axis is expressed as '5.56e+07'.
So, I want to know if there is a method in finplot that can change the unit of the Y axis.
If is not, I will change the Y-axis by adding the new data in the dataframe.

Tell me the method and I'll find it.
Because your time is very valuable.

I don't speak English. So I borrowed the power of Google Translate.

Thanks for reading.

And your finplot is very amazing.

**Reply by highfestiva** [reply]
I can't tell for sure, as your example is *not complete*, but I'm guessing you should do something like
```python
self.chart_ma20.update_gfx()
...
```
after `fplt._repaint_candles()`. See `examples/complicated.py` which you can copy-paste from.

**Reply by youmeomeyou** [reply]
Thank you for your reply.

update_gfx() is correct.

and

Finally I found the cause.
When the data on the Y-axis exceeds 100000, the data on the Y-axis is expressed as '3.9e + 06'.
This is because our currency unit is 1000 times larger than the dollar.

Thank you very much for your kind reply.
And the finplot you made is really good.

**Reply by highfestiva** [reply]
Thanks, good luck!
